packages/terra-table/src/subcomponents/ColumnHeaderCell.jsx

```diff
@@ -0,0 +1,220 @@
+import React, { useContext, useRef, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { injectIntl } from 'react-intl';
+import * as KeyCode from 'keycode-js';
+import classNames from 'classnames/bind';
+
+import ThemeContext from 'terra-theme-context';
+import IconUp from 'terra-icon/lib/icon/IconUp';
+import IconDown from 'terra-icon/lib/icon/IconDown';
+import IconError from 'terra-icon/lib/icon/IconError';
+
+import ColumnResizeHandle from './ColumnResizeHandle';
+import { SortIndicators } from '../proptypes/columnShape';
+import ColumnContext from '../utils/ColumnContext';
+import styles from './ColumnHeaderCell.module.scss';
+import GridContext from '../utils/GridContext';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Required string representing a unique identifier for the column header cell.
+   */
+  id: PropTypes.string.isRequired,
+
+  /**
+   * String of text to render within the column header cell.
+   */
+  displayName: PropTypes.string,
+
+  /**
+   * A string indicating which sorting indicator should be rendered. If not provided, no sorting indicator will be rendered.
+   * If a `component` value is specified, `sortIndicator` will be ignored. One of `ascending`, `descending`.
+   */
+  sortIndicator: PropTypes.oneOf(Object.values(SortIndicators)),
+
+  /**
+   * Boolean value indicating whether or not the column has an error in the data.
+   */
+  hasError: PropTypes.bool,
+
+  /**
+   * Number that specifies the minimum column width in pixels.
+   */
+  minimumWidth: PropTypes.number,
+
+  /**
+   * Number that specifies the maximum column width in pixels.
+   */
+  maximumWidth: PropTypes.number,
+
+  /**
+   * Boolean value indicating whether or not the column header is selectable.
+  */
+  isSelectable: PropTypes.bool,
+
+  /**
+   * Boolean value indicating whether or not the column header is resizable.
+   */
+  isResizable: PropTypes.bool,
+
+  /**
+   * Height of the parent table.
+   */
+  tableHeight: PropTypes.number,
+
+  /**
+   * String that specifies the default width for columns in the table. Any valid CSS width value is accepted.
+   */
+  width: PropTypes.number.isRequired,
+
+  /**
+   * String that specifies the column height. Any valid CSS height value accepted.
+  */
+  headerHeight: PropTypes.string.isRequired,
```

Comment:
Can we put all required tops at the top?

Reply:
Done